addons/web/static/src/core/datetime/datetimepicker_service.js

```diff
@@ -77,18 +77,22 @@ export const datetimePickerService = {
                 });
                 // Hook methods
 
+                let lastAppliedValue = undefined;
                 /**
                  * Wrapper method on the "onApply" callback to only call it when the
                  * value has changed, and set other internal variables accordingly.
                  */
                 const apply = () => {
-                    if (areDatesEqual(lastInitialProps?.value, deepCopy(pickerProps.value))) {
+                    const valueCopy = deepCopy(pickerProps.value);
+                    if (areDatesEqual(lastInitialProps?.value, valueCopy) ||
+                        areDatesEqual(lastAppliedValue, valueCopy)) {
```

Comment:
I'd rather only keep `lastAppliedValue` as the reference for the last applied value, and *also* update it at the same time as `lastInitialProps` in `computeBasePickerProps` (line ~106)
```suggestion
                    if (areDatesEqual(lastAppliedValue, valueCopy)) {
```